### PR TITLE
fix(receipts): Remove purchase receipt view & refactor payment status display

### LINKED ISSUE #477: User shouldn't be able to view a purchase receipt when it looks so similar to a swipe receipt

### PR DESCRIPTION
**Pull request by marfavi**
Closes #477.
- `PurchaseReceiptListEntry` widget now has 'tappable' set to false, disabling purchase receipts.
- Timestamp has been added below the purchase/swipe status in `ReceiptListEntry`.
- Updated `PaymentStatus` string values to provide clearer descriptions.
- Renamed 'purchaseStatus' field to 'status' across multiple generic receipt widgets.
- Refactored `PurchaseReceiptListEntry` to show price differently (or hide it), based on status.
- misc: Updated `_formatter` to `_formatDate`/`_formatDateTime` in `ReceiptListEntry` and `ReceiptCard`.
- misc: Removed rounded edges from `ReceiptListEntry`.

### CODE REVIEW

File: lib/features/purchase/domain/entities/payment_status.dart [hunk 1]
```diff
@@ -32,12 +32,12 @@ enum PaymentStatus {
   @override
   String toString() {
     return switch (this) {
-      PaymentStatus.completed => 'Completed',
-      PaymentStatus.rejectedPayment => 'Rejected',
-      PaymentStatus.awaitingPayment => 'Pending',
-      PaymentStatus.refunded => 'Refunded',
-      PaymentStatus.error => 'Error',
-      PaymentStatus.reserved => 'Reserved',
+      PaymentStatus.completed => 'Purchased',
```
Comment (fremartini) [inline lib/features/purchase/domain/entities/payment_status.dart:35]: perhaps these should be in Strings.dart

File: lib/features/receipt/presentation/widgets/list_entry/purchase_receipt_list_entry.dart [hunk 2]
```diff
@@ -10,18 +11,32 @@ class PurchaseReceiptListEntry extends StatelessWidget {
     required this.receipt,
   });
 
+  String get priceText {
+    final price = receipt.price;
+    return switch (receipt.paymentStatus) {
+      PaymentStatus.completed => '$price,-',
```
Comment (fremartini) [inline lib/features/receipt/presentation/widgets/list_entry/purchase_receipt_list_entry.dart:17]: perhaps we should include a function formatPrice in strings.dart that handles formatting currency related strings?

File: lib/features/receipt/presentation/widgets/list_entry/purchase_receipt_list_entry.dart [hunk 3]
```diff
@@ -10,18 +11,32 @@ class PurchaseReceiptListEntry extends StatelessWidget {
...
+        '($price,-)',
+      PaymentStatus.error || PaymentStatus.rejectedPayment => '',
+    };
+  }
+
   @override
   Widget build(BuildContext context) {
     return ReceiptListEntry(
-      tappable: true,
+      tappable: false,
```
Comment (fremartini) [inline lib/features/receipt/presentation/widgets/list_entry/purchase_receipt_list_entry.dart:29]: how come we no longer want to be able to open a purchase? this removes users ability to view the exact timestamp of the purchase that is displayed in the opened receipt

**Review by fremartini — CHANGES_REQUESTED** [review]
LGTM but we need to clarify why Purchases no longer need to be able to be opened

File: lib/features/receipt/presentation/widgets/list_entry/purchase_receipt_list_entry.dart (hunk 3 above)
Comment (marfavi) [reply lib/features/receipt/presentation/widgets/list_entry/purchase_receipt_list_entry.dart:29]: #477. The "receipt" of a purchase (especially incomplete purchases) should never resemble like a swipe receipt.

At some point we should add the purchase receipt page as specified in the v4 design spec (screenshot below)
image

Comment (fremartini) [reply lib/features/receipt/presentation/widgets/list_entry/purchase_receipt_list_entry.dart:29]: i agree, but as it stands it now removes users ability to view the exact timestamp of the purchase. Might i suggest we disable this feature when migrating to the new design?

Comment (marfavi) [reply lib/features/receipt/presentation/widgets/list_entry/purchase_receipt_list_entry.dart:29]: Then we'll just add the timestamp to the receipt list entry for now. See latest commit

File: lib/features/purchase/domain/entities/payment_status.dart (hunk 1 above)
Comment (marfavi) [reply lib/features/purchase/domain/entities/payment_status.dart:35]: Might agree, but out of scope of this PR

File: lib/features/receipt/presentation/widgets/list_entry/purchase_receipt_list_entry.dart (hunk 2 above)
Comment (marfavi) [reply lib/features/receipt/presentation/widgets/list_entry/purchase_receipt_list_entry.dart:17]: Might agree, but out of scope of this PR

Comment (fremartini) [reply lib/features/receipt/presentation/widgets/list_entry/purchase_receipt_list_entry.dart:17]: there is already a price function in Strings.dart l. 159. I would prefer if this was fixed now while we are fixing stuff here

File: lib/features/purchase/domain/entities/payment_status.dart (hunk 1 above)
Comment (fremartini) [reply lib/features/purchase/domain/entities/payment_status.dart:35]: I would prefer if this was fixed now while we are fixing stuff here

**Review by fremartini — CHANGES_REQUESTED** [review]
Please address my previous comments regarding Strings.dart

**Review by fremartini — APPROVED** [review]

**Pull request merged**